### PR TITLE
Fixes #33040 - Enables upgrade-all via agent actions

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by parthaa** [comment]
Steps to test:

- Have a katello-agent enabled  content host that is feeding off https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/
- `yum install walrus`
- `yum downgrade walrus`
- Now go to the Package Actions on the  Content Host in the UI
- Click on `Update All Packages`

Before PR the walrus rpm will not get updated
```bash
# rpm -q walrus
walrus-0.71-1.noarch
```
After PR 
```bash
# rpm -q walrus
walrus-5.21-1.noarch
```

File: app/lib/katello/agent/base_message.rb [hunk 1]
```diff
@@ -30,6 +27,12 @@ def json
         }
       end
 
+      def options
```
Comment (jeremylenz) [inline app/lib/katello/agent/base_message.rb:30]: May be helpful to put a comment here. I could see myself wondering "why does this need to be a method?"
```suggestion
      def options # may be overridden by children
```

**Review by jeremylenz — COMMENTED** [review]
Thanks @parthaa!

Testing steps work exactly as expected :+1: 

I don't see any test coverage for this area, might be nice to have some?

**Comment by parthaa** [comment]
> I don't see any test coverage for this area, might be nice to have some?

added tests

**Comment by jturel** [comment]
I think this originally should have been implemented as a separate action -> https://github.com/Katello/katello/blob/ea44c8eab2597b6234f1a4b0efbb1bd6cd7e4ffc/app/controllers/katello/api/v2/host_packages_controller.rb#L62

Notice that update and update_all share the same dynflow action. Separating those would mean there could be a 'UpgradeAllPackagesMessage'. Just sharing what a better design(in the past) would have been. Please don't change the code to do this - it's already living on borrowed time.

**Comment by parthaa** [comment]
> I think this originally should have been implemented as a separate action

thats a neat idea.

**Comment by parthaa** [comment]
[test katello]

File: test/lib/agent/update_package_message_test.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require 'katello_test_helper'
+
+module Katello
+  module Agent
+    class UpdatePackageMessageTest < ActiveSupport::TestCase
+      def test_all_package_update
+        json = UpdatePackageMessage.new(content: [], consumer_id: "100000").json
+        assert json[:request][:args][1][:all]
```
Comment (jeremylenz) [inline test/lib/agent/update_package_message_test.rb:8]:
```suggestion
        options = json[:request][:args][1]
        assert options[:all]
```

File: test/lib/agent/update_package_message_test.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        assert_empty json[:request][:args][0][0][:unit_key]
+      end
+
+      def test_content_package_update
+        json = UpdatePackageMessage.new(content: ["foo.rpm"], consumer_id: "100000").json
+        assert_nil json[:request][:args][1][:all]
```
Comment (jeremylenz) [inline test/lib/agent/update_package_message_test.rb:14]: Works fine, but might be nice to use the same variable names as the corresponding code so it's a bit easier to see what's going on:
```suggestion
        options = json[:request][:args][1]
        units = json[:request][:args][0]
        assert_nil options[:all]
        assert_equal "foo.rpm", units[0][:unit_key][:name]
```

**Review by jeremylenz — APPROVED** [review]
Thanks @parthaa!

I'll APJ, but also leave a small suggestion about test readability below.

**Pull request merged**